src/test/java/org/jenkinsci/plugins/ewm/steps/ExwsAllocateStepTest.java

```diff
@@ -0,0 +1,247 @@
+import java.util.Arrays;
+import java.util.Collections;
+
+import static hudson.model.Result.FAILURE;
+import static java.lang.String.format;
+import static org.apache.commons.lang.RandomStringUtils.randomAlphanumeric;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.jenkinsci.plugins.ewm.TestUtil.*;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Unit tests for {@link ExwsAllocateStep}.
+ *
+ * @author Alexandru Somai
+ */
+public class ExwsAllocateStepTest {
+
+    @ClassRule
+    public static JenkinsRule j = new JenkinsRule();
+    @ClassRule
+    public static TemporaryFolder tmp1 = new TemporaryFolder();
+    @ClassRule
+    public static TemporaryFolder tmp2 = new TemporaryFolder();
+
+    private static File pathToDisk1;
+    private static File pathToDisk2;
+
+    private WorkflowRun upstreamRun;
+    private WorkflowRun downstreamRun;
+
+    @BeforeClass
+    public static void setUpPathToDisks() throws IOException {
+        pathToDisk1 = tmp1.newFolder("mount-to-disk-one");
+        pathToDisk2 = tmp2.newFolder("mount-to-disk-two");
+    }
+
+    @After
+    public void tearDown() {
+        setUpDiskPools(j.jenkins, Collections.<DiskPool>emptyList());
+    }
+
+    /* ##### Tests for the upstream Job ###### */
+
+    @Test
+    public void missingDiskPoolId() throws Exception {
+        createUpstreamJobAndRun(" ");
+
+        j.assertBuildStatus(FAILURE, upstreamRun);
+        j.assertLogContains("Disk Pool ID was not provided as step parameter", upstreamRun);
+    }
+
+    @Test
+    public void wrongDiskPoolId() throws Exception {
+        setUpDiskPool();
+        createUpstreamJobAndRun(DISK_POOL_ID + "random");
+
+        j.assertBuildStatus(FAILURE, upstreamRun);
+        j.assertLogContains(format("No Disk Pool ID matching '%s' was found in the global config", DISK_POOL_ID + "random"), upstreamRun);
+    }
+
+    @Test
+    public void emptyDisks() throws Exception {
+        setUpDiskPool();
+        createUpstreamJobAndRun();
+
+        j.assertBuildStatus(FAILURE, upstreamRun);
+        j.assertLogContains(format("No Disks were defined in the global config for Disk Pool ID '%s'", DISK_POOL_ID), upstreamRun);
+    }
+
+    @Test
+    public void missingDiskId() throws Exception {
+        setUpDiskPool(new Disk("", "name", "mount", "path"));
+        createUpstreamJobAndRun();
+
+        j.assertBuildStatus(FAILURE, upstreamRun);
+        j.assertLogContains(format("Disk ID was not provided in the Jenkins global config for the Disk Pool ID '%s'", DISK_POOL_ID), upstreamRun);
+    }
+
+    @Test
+    public void missingMasterMountPoint() throws Exception {
+        setUpDiskPool(new Disk(DISK_ID_ONE, "name", "", "path"));
+        createUpstreamJobAndRun();
+
+        j.assertBuildStatus(FAILURE, upstreamRun);
+        j.assertLogContains(format("Mounting point from Master to the disk is not defined for Disk ID '%s', from Disk Pool ID '%s'", DISK_ID_ONE, DISK_POOL_ID), upstreamRun);
+    }
```

Comment:
I think it is okay to have the `missingMasterMountPoint` test here for now, but when there is more than one allocation strategy, it may be better to place strategy testing in a test file of its own. At this point, no change is needed IMO.